### PR TITLE
Item 6199: support other assay provider types in AssayDesignerPanels and AssayPropertiesPanel

### PR DESCRIPTION
**Pull request by cnathe**

### CODE REVIEW

File: assay/src/client/AssayDesigner/AssayDesigner.tsx [hunk 1]
```diff
@@ -38,18 +40,26 @@ export class App extends React.Component<any, State> {
     {
         super(props);
 
-        const { rowId, returnUrl } = ActionURL.getParameters();
+        const { rowId, copy, providerName } = ActionURL.getParameters();
+
+        // hack, if the returnUrl has stripped off the rowId because of encoding/decoding issues (see TODO in AbstractAssayProvider.getManageMenuNavTree()) add it back on
+        let returnUrl = ActionURL.getParameter('returnUrl');
+        if (rowId && returnUrl && returnUrl.indexOf('rowId') === returnUrl.length - 5) {
```
Comment (labkey-martyp) [inline assay/src/client/AssayDesigner/AssayDesigner.tsx]: Not sure if rowId could be zero here and mess up this check.

File: assay/src/client/AssayDesigner/AssayDesigner.tsx [hunk 2]
```diff
@@ -167,7 +207,7 @@ export class App extends React.Component<any, State> {
         }
 
         // check if this is a create assay case with a user that doesn't have permissions
-        if (model === undefined && !hasDesignAssayPerm) {
+        if (!model.protocolId && !hasDesignAssayPerm) {
```
Comment (labkey-martyp) [inline assay/src/client/AssayDesigner/AssayDesigner.tsx]: Possibly protocolId is zero?

File: assay/src/org/labkey/assay/AssayController.java [hunk 3]
```diff
@@ -1552,4 +1554,26 @@ public NavTree appendNavTrail(NavTree root)
...
     }
+
+    @RequiresPermission(ReadPermission.class)
+    public class GetValidPublishTargetsAction extends ReadOnlyApiAction
+    {
+        @Override
+        public ApiResponse execute(Object object, BindException errors)
+        {
+            ApiSimpleResponse response = new ApiSimpleResponse();
+            List<Map<String, Object>> containersInfo = new ArrayList();
```
Comment (labkey-martyp) [inline assay/src/org/labkey/assay/AssayController.java]: new ArrayList<>() ?

File: assay/src/org/labkey/assay/AssayController.java [hunk 4]
```diff
@@ -1552,4 +1554,26 @@ public NavTree appendNavTrail(NavTree root)
...
+            for (Study study : AssayPublishService.get().getValidPublishTargets(getUser(), ReadPermission.class))
+            {
+                Container container = study.getContainer();
+                Map<String, Object> containerInfo = new HashMap();
```
Comment (labkey-martyp) [inline assay/src/org/labkey/assay/AssayController.java]: new HashMap<>() ?

File: assay/src/client/AssayDesigner/AssayDesigner.tsx (hunk 1 above)
Comment (cnathe) [reply assay/src/client/AssayDesigner/AssayDesigner.tsx]: Good call. I don't think zero as a rowId is valid in this case, but easy enough to just change the check to `rowId !== undefined`. Done.

File: assay/src/client/AssayDesigner/AssayDesigner.tsx (hunk 2 above)
Comment (cnathe) [reply assay/src/client/AssayDesigner/AssayDesigner.tsx]: I don't think zero is valid for protocolId here, but there is a better check to use: `model.isNew()`. Changed.

File: assay/src/org/labkey/assay/AssayController.java (hunk 3 above)
Comment (cnathe) [reply assay/src/org/labkey/assay/AssayController.java]: Yes. Fixed.

File: assay/src/org/labkey/assay/AssayController.java (hunk 4 above)
Comment (cnathe) [reply assay/src/org/labkey/assay/AssayController.java]: Yes. Fixed.

**Pull request merged**